src/Native/Unix/System.Native/pal_io.cpp

```diff
@@ -1171,3 +1171,22 @@ extern "C" char* SystemNative_RealPath(const char* path)
     assert(path != nullptr);
     return realpath(path, nullptr);
 }
+
+extern "C" int32_t SystemNative_LockFileRegion(int fd, int64_t offset, int64_t length, int16_t lockType)
```

Comment:
Shouldn't be using `int` in PAL signatures either. All of the file descriptors were made intptr_t so that they can can be marshalled as safehandles.